Fixes for CHIPS and cookies with __Host_ prefix (#34547)
mdn/content: files/en-us/web/privacy/privacy_sandbox/partitioned_cookies/index.md

Without cookie partitioning, third-party cookies can enable services to track users and associate their information across unrelated top-level sites. Cookies marked `Partitioned` are double-keyed: by the origin that sets them _and_ the origin of the top-level page.

> This means they can only be read within the context of the top-level site they were set on. This allows cross-site tracking to be blocked, while still enabling legitimate uses of third-party cookies such as persisting state of embedded maps or chat widgets across a domain and its subdomains, and persisting config information for subresource CDN load balancing and Headless CMS providers.

Author: This was incorrect - if you partition the cookie explicitly can't be sent to other sites. It is shared by the site and its subdomains.

Reviewer: Yes, agreed.

Set-Cookie: __Host-example=34d8g; SameSite=None; Secure; Path=/; Partitioned;
```

> > **Note:** Partitioned cookies must be set with `Secure`. In addition, you could use the `__Host` prefix when setting partitioned cookies to bind them only to the current host and not any other subdomains (or the domain if you were binding to a subdomain).

Author: The information about setting the `__Host` prefix was a little unclear (to me). As per https://github.com/mdn/content/pull/34547#pullrequestreview-2152506991 using this means that whatever host subdomain or domain set the cookie is the only thing that can read it.

Recommending to use this is wrong! If you do this, then the main use case of partitions - sharing among subdomains - is prevented.

But I am not sure that we shouldn't explicitly say "don't use this with partitioned".

Why? Because if you set host prefix then the cookie is never shared whether you're partitioned or not. So it is redundant to partition the cookie as well.

@chrisdavidmills What am I missing here/When would we use `__Host`?

Reviewer: AFAIUI:

- `Partitioned` means that the 3rd party cookie can be shared with the host it is embedded on, and any other subdomains of the domain that host resides on.
- `Partitioned` + `__Host-` means that the 3rd party cookie can be shared only with the host it is embedded on, not other subdomains. 

So, if you want maximum privacy and only want to share it with that specific embedding host, then the latter s the way to go. This is also what the Google docs on it seem to say, for example https://developers.google.com/privacy-sandbox/3pcd/chips

Author: @chrisdavidmills Thank you, but I don't think I was clear. Below are your to options plus a third option. 

- `Partitioned` means that the 3rd party cookie can be shared with the host it is embedded on, and any other subdomains of the domain that host resides on.
- `Partitioned` + `__Host-` means that the 3rd party cookie can be shared only with the host it is embedded on, not other subdomains. 
- `__Host-` (without Partitioned) means that the 3rd party cookie can be shared only with the host it is embedded on, not other domains or subdomains. 

So 
1. what's the difference between the last two options? In both cases you have the same same restriction, so why do you need to set Partitioned?
2. IFF you set the host on a partitioned cookie you are restricting to just that that host, which is the most secure/restrictive. But my take on this is that the advice to set this is really rubbish because at the same time you've also removed the main benefits of partitioning, which is to allow cross-subdomain sharing but not cross-domain sharing.

I can't think of a reason to do this, other than perhaps that in future Chrome intends to _only_ allow Partitioned third party cookies some point soon. At that point, even though they appear the same, just having `__Host_` would not be enough?
If so, we should say that?

Is there any way we can find out?

Author: @chrisdavidmills FYI I changed this to

```
> **Note:** Partitioned cookies must be set with `Secure`. In addition, you can use the `__Host` prefix when setting partitioned cookies to bind them only to the current domain or subdomain, and this is recommended if you don't need to share cookies between subdomains.
```

The text is AFAIK accurate, but if we can confirm the reason why you can't just use `__Host_` we will need to modify it again. Specifically, we'd want to update the `__Host_` text everywhere to note that you should call it with `Partitioned` (assuming that I'm right, and in future if you don't have Partitioned the cookie will be blocked _even though it is safe_.

Reviewer: Thanks for the clarification, @hamishwillee. I think these are valid concerns, and I don't really have an answer. Summoning @samdutton and @mihajlija to see if they have any insights/answers (Hi folks!)

> > **Note:** Partitioned cookies must be set with `Secure`. In addition, you could use the `__Host` prefix when setting partitioned cookies to bind them only to the current host and not any other subdomains (or the domain if you were binding to a subdomain).

Author: ```suggestion
> **Note:** Partitioned cookies must be set with `Secure`. In addition, you can use the `__Host` prefix when setting partitioned cookies to bind them only to the current domain or subdomain, and this is recommended if you don't need to share cookies between subdomains.
```

> > **Note:** Partitioned cookies must be set with `Secure`. In addition, you could use the `__Host` prefix when setting partitioned cookies to bind them only to the current host (domain or subdomain) and not any other subdomains (though in this case there would not be much need for a partitioned cookie).

Author: ```suggestion
> **Note:** Partitioned cookies must be set with `Secure`. In addition, you could use the `__Host` prefix when setting partitioned cookies to bind them only to the current host and not any other subdomains (or the domain if you were binding to a subdomain).
```

2. The storage key for the cookie would be `{("https://shoppy.example"), ("3rd-party.example/chat")}`.
3. The user visits various subdomains that also embed `https://3rd-party.example/chat`, including `https://support.shoppy.example` and `https://checkout.shoppy.example`. The new embedded instances are able to access the cookie because the partition key still matches.

> You can restrict the cookie to just a domain and a particular subdomain by setting the cookie with the `__Host` prefix.
> 

Author: Removing. Don't know why you would do this
```suggestion
```
